torch_geometric/data/data.py

```diff
@@ -516,34 +516,43 @@ def to_heterogeneous(self, node_type: Optional[Tensor] = None,
                 edge_type_names.append((node_type_names[src_types[0]], str(i),
                                         node_type_names[dst_types[0]]))
 
+        # iterate over node types to get node slices
+        # and index_map, which contains the index of each node
+        # in the heterogenous graph.
+        node_slices, index_map = {}, torch.zeros_like(node_type)
+        for i, key in enumerate(node_type_names):
+            node_slices[i] = torch.where(node_type == i)[0]
```

Comment:
```suggestion
            node_slices[i] = (node_type == i).nonzero(as_tuple=False).view(-1)
```
I find this easier to read, but I guess this depends :)